runtime/lib/backend/null_qubit/NullQubit.hpp

```diff
@@ -56,6 +113,10 @@ struct NullQubit final : public Catalyst::Runtime::QuantumDevice {
     auto AllocateQubit() -> QubitIdType
     {
         num_qubits_++; // next_id
+        if (this->track_resources_) {
```

Comment:
You also need to update `AllocateQubits` and `ReleaseQubit` (and `ReleaseQubits`)?

Reply:
`AllocateQubits` calls `AllocateQubit`. As for Release, I was taking "num_qubits" to mean the maximum allocated amount at any point since device creation. Since we print in the destructor, changing the qubit number on release would always track as "0". Perhaps this could be avoided by taking @dime10 's suggestion of printing in Release instead of the destructor